src/nimiq_utils.c

```diff
@@ -205,6 +210,22 @@ void print_network_id(uint8_t *in, char *out) {
     }
 }
 
+void print_extra_data(uint8_t *in, char *out, uint16_t data_size) {
+    // Make sure that the string is always null-terminated
+    out[MAX_DATA_LENGTH] = '\0';
+
```

Comment:
Please add a safety check to ensure that data_size < MAX_DATA_LENGTH, and throw an error otherwise.